src/Microsoft.AspNetCore.Server.IntegrationTesting/Deployers/IISExpressDeployer.cs

```diff
@@ -42,15 +42,79 @@ public IISExpressDeployer(DeploymentParameters deploymentParameters, ILoggerFact
                 // Start timer
                 StartTimer();
 
-                // For now we always auto-publish. Otherwise we'll have to write our own local web.config for the HttpPlatformHandler
-                DotnetPublish();
+                var targetFramework = DeploymentParameters.TargetFramework;
 
-                var contentRoot = DeploymentParameters.PublishedApplicationRootPath;
+                // IISIntegration uses this layout
+                var dllRoot = Path.Combine(DeploymentParameters.ApplicationPath, "bin", DeploymentParameters.RuntimeArchitecture.ToString(),
+                    DeploymentParameters.Configuration, targetFramework);
+
+                if (!Directory.Exists(dllRoot))
+                {
+                    // Most repos use this layout
+                    dllRoot = Path.Combine(DeploymentParameters.ApplicationPath, "bin", DeploymentParameters.Configuration, targetFramework);
+
+                    if (!Directory.Exists(dllRoot))
+                    {
+                        // The bits we need weren't pre-compiled, compile on publish
```

Comment:
Where are application published to on pre-compilation? Temp or app local?

Reply:
Local, and it's not a proper publish, only a compile.